### DISCUSSION TITLE
Converting integer terms to real in the C++ API

### DISCUSSION THREAD

**Discussion by nicola-gigante** [cvc5/cvc5#9179] [Q&A] [answered]
How can I convert a term `t1` of integer sort into an equivalent real-sorted term, in the C++ API?

I mean, something equivalent of `(to_real t1)` in `smtlib2`, or similar to `Z3_mk_int2real()` in the Z3 API.

Is there anything like that in cvc5?

Thank you very much,
Nicola

**Comment by aniemetz** [comment, marked as answer]
You can use the `TO_REAL` operator for this:
https://github.com/cvc5/cvc5/blob/main/src/api/cpp/cvc5_kind.h#L1042

**Reply by nicola-gigante** [reply]
Ah, yes! I don't know how I missed it. Thanks!
